tests/LaminasIntegration/data/pluginMethodDynamicReturn.php

```diff
@@ -14,10 +15,22 @@ final class pluginMethodDynamicReturn
      */
     private $phpRenderer;
 
-    public function getDynamicType(): void
+    public function getDynamicTypeFromStaticString(): void
     {
         $urlHelper = $this->phpRenderer->plugin('url');
         $urlHelper->setRouter(new SimpleRouteStack());
         $urlHelper->foobar();
     }
+
+    public function callGetDynamicTypeFromVariable(): void
+    {
+        $urlHelper = $this->getDynamicTypeFromStringVariable('url');
+        $urlHelper->foobar();
+    }
+
+    public function getDynamicTypeFromStringVariable(string $name): AbstractHelper
+    {
+        return $this->phpRenderer->plugin($name);
```

Comment:
@ondrejmirtes we would like to catch here the raw value of `$name` variable, which is `'url'` from few lines above, can we?

Reply:
You can see the literal values only in the same method body, not across method boundaries.

Reply:
Ok thank you